pkg/infra/log/slog.go

```diff
@@ -0,0 +1,59 @@
+package log
+
+import (
+	"context"
+	"log/slog"
+)
+
+var _ slog.Handler = &SLogHandler{}
+
+type SLogHandler struct {
+	Logger
+}
+
+// NewSLogHandler returns a SLogHandler with the given level.
+func NewSLogHandler(logger Logger) *SLogHandler {
+	return &SLogHandler{logger}
+}
+
+func (h *SLogHandler) Enabled(_ context.Context, level slog.Level) bool {
+	return true
+}
+
+// Handle implements Handler.Handle.
+func (h *SLogHandler) Handle(ctx context.Context, r slog.Record) error {
+	attrs := []any{}
+	fn := func(attr slog.Attr) bool {
+		attrs = append(attrs, attr.Key, attr.Value)
+		return true
+	}
+	r.Attrs(fn)
+	switch r.Level {
+	case slog.LevelDebug:
+		h.Debug(r.Message, attrs...)
+	case slog.LevelInfo:
+		h.Info(r.Message, attrs...)
+	case slog.LevelWarn:
+		h.Warn(r.Message, attrs...)
+	case slog.LevelError:
+		h.Error(r.Message, attrs...)
+	}
+	return nil
+}
+
+// WithAttrs implements Handler.WithAttrs.
+func (h *SLogHandler) WithAttrs(attrs []slog.Attr) slog.Handler {
+	out := []any{}
```

Comment:
You can likely preallocate with `2*len(attrs)`: similar to [here](https://github.com/kubernetes/klog/blob/main/internal/sloghandler/sloghandler_slog.go#L81)

Reply:
good call. updated